pkg/apicheck/check.go

```diff
@@ -129,12 +130,17 @@ func (c *ApiConnectivityCheck) getWorkerPeersResponse() peers.Response {
 
 	c.config.Log.Info("Error count exceeds threshold, trying to ask other nodes if I'm healthy")
 	peersToAsk := c.config.Peers.GetPeersAddresses(peers.Worker)
-	if peersToAsk == nil || len(peersToAsk) == 0 {
-		c.config.Log.Info("Peers list is empty and / or couldn't be retrieved from server, nothing we can do, so consider the node being healthy")
-		// TODO: maybe we need to check if this happens too much and reboot
+	if peersToAsk == nil && c.config.MinPeersForRemediation != 0 || len(peersToAsk) < c.config.MinPeersForRemediation {
```

Comment:
It's a bit tricky, but as `len(peersToAsk)` is zero if `peersToAsk` is `nil`, I think you can get rid of the first part and just use `len(peersToAsk) < c.config.MinPeersForRemediation`.


* if `peersToAsk == nil` (and so `len(...) == 0`) and `c.config.MinPeersForRemediation != 0`, then also `len(peersToAsk) < c.config.MinPeersForRemediation` is True.
* For all the other combinations, we always need to evaluate the part after `||` anyway

Reply:
We did update this to try and make it more clear, and added more comments - and removed an unnecessary check.